File: Codes/3226-number-of-bit-changes-to-make-two-integers-equal-WeYunSuan.py
'''
Author: LetMeFly
Date: 2024-11-02 10:56:41
LastEditors: LetMeFly.xyz
LastEditTime: 2024-11-02 10:58:47
'''
class Solution:
    def minChanges(self, n: int, k: int) -> int:
        return (n ^ k).bit_count() if (n | k) == n else -1

File: Codes/3226-number-of-bit-changes-to-make-two-integers-equal-WeYunSuan.java
/*
 * @Author: LetMeFly
 * @Date: 2024-11-02 10:59:45
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2024-11-02 10:59:53
 */
class Solution {
    public int minChanges(int n, int k) {
        return (n | k) == n ? Integer.bitCount(n ^ k) : -1;
    }
}